components/places/src/bookmark_sync/engine.rs

```diff
@@ -61,6 +62,93 @@ where
     }
 }
 
+// `SyncEngine` implementation for bookmarks
+pub struct BookmarksSyncEngine {
```

Comment:
Can we just change the existing engine to not use lifetimes? Probably `Arc<PlacesApi>` rather than `Arc<PlacesDb>`, but that's a minor detail (and possibly wrong :) That could probably be done independently of uniffi'ing the sync manager, but it makes sense to me to do that first rather than leaving this massive "todo" which we might never do - there's no good reason to not take this refactor on now given there's no urgency on the uniffi of the sync manager. FWIW, I'd be very surprised if this refactor ends up with more than trivial conflicts from the unification of places.

Reply:
> Can we just change the existing engine to not use lifetimes? Probably `Arc<PlacesApi>` rather than `Arc<PlacesDb>`, but that's a minor detail (and possibly wrong :) That could probably be done independently of uniffi'ing the sync manager, but it makes sense to me to do that first rather than leaving this massive "todo" which we might never do - there's no good reason to not take this refactor on now given there's no urgency on the uniffi of the sync manager. FWIW, I'd be very surprised if this refactor ends up with more than trivial conflicts from the unification of places.

Totally on board with this one now, based on:
  - Moving iOS to using the sync manager isn't trivia, therefore we need to continue to support the `PlacesApi` sync methods for a while
  - We can delete the import functionality